langchain/src/vectorstores/prisma.ts

```diff
@@ -410,9 +412,18 @@ export class PrismaVectorStore<
         Object.entries(ops).map(([opName, value]) => {
           // column name, operators cannot be parametrised
           // these fields are thus not escaped by Prisma and can be dangerous if user input is used
+          const opNameKey = opName as keyof typeof OpMap;
           const colRaw = this.Prisma.raw(`"${key}"`);
-          const opRaw = this.Prisma.raw(OpMap[opName as keyof typeof OpMap]);
-          return this.Prisma.sql`${colRaw} ${opRaw} ${value}`;
+          const opRaw = this.Prisma.raw(OpMap[opNameKey]);
+
+          switch (OpMap[opNameKey]) {
+            case OpMap.in:
+              return this.Prisma.sql`${colRaw} ${opRaw} (${(
+                value as string[]
+              ).join(",")})`;
```

Comment:
Instead of casting could we instead use a typeguard? Something like this
```suggestion
          const opRaw = this.Prisma.raw(OpMap[opNameKey]);

          switch (OpMap[opNameKey]) {
            case OpMap.in: {
              if (!Array.isArray(values) || !values.every((v) => typeof v === "string")) {
                throw new Error("...")
              }
              return this.Prisma.sql`${colRaw} ${opRaw} (${(
                value as string[]
              ).join(",")})`;
            }
```